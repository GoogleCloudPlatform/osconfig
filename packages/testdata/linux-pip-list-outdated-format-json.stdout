[{"name": "attrs", "version": "19.3.0", "latest_version": "25.3.0", "latest_filetype": "wheel"}, {"name": "Automat", "version": "0.8.0", "latest_version": "24.8.1", "latest_filetype": "wheel"}, {"name": "blinker", "version": "1.4", "latest_version": "1.8.2", "latest_filetype": "wheel"}, {"name": "certifi", "version": "2019.11.28", "latest_version": "2025.4.26", "latest_filetype": "wheel"}, {"name": "chardet", "version": "3.0.4", "latest_version": "5.2.0", "latest_filetype": "wheel"}, {"name": "Click", "version": "7.0", "latest_version": "8.1.8", "latest_filetype": "wheel"}, {"name": "colorama", "version": "0.4.3", "latest_version": "0.4.6", "latest_filetype": "wheel"}, {"name": "configobj", "version": "5.0.6", "latest_version": "5.0.9", "latest_filetype": "wheel"}, {"name": "constantly", "version": "15.1.0", "latest_version": "23.10.4", "latest_filetype": "wheel"}, {"name": "cryptography", "version": "2.8", "latest_version": "44.0.3", "latest_filetype": "wheel"}, {"name": "dbus-python", "version": "1.2.16", "latest_version": "1.4.0", "latest_filetype": "sdist"}, {"name": "distro", "version": "1.4.0", "latest_version": "1.9.0", "latest_filetype": "wheel"}, {"name": "distro-info", "version": "0.23+ubuntu1.1", "latest_version": "1.0", "latest_filetype": "wheel"}, {"name": "entrypoints", "version": "0.3", "latest_version": "0.4", "latest_filetype": "wheel"}, {"name": "httplib2", "version": "0.14.0", "latest_version": "0.22.0", "latest_filetype": "wheel"}, {"name": "hyperlink", "version": "19.0.0", "latest_version": "21.0.0", "latest_filetype": "wheel"}, {"name": "idna", "version": "2.8", "latest_version": "3.10", "latest_filetype": "wheel"}, {"name": "importlib-metadata", "version": "1.5.0", "latest_version": "8.5.0", "latest_filetype": "wheel"}, {"name": "incremental", "version": "16.10.1", "latest_version": "24.7.2", "latest_filetype": "wheel"}, {"name": "Jinja2", "version": "2.10.1", "latest_version": "3.1.6", "latest_filetype": "wheel"}, {"name": "jsonpatch", "version": "1.22", "latest_version": "1.33", "latest_filetype": "wheel"}, {"name": "jsonpointer", "version": "2.0", "latest_version": "3.0.0", "latest_filetype": "wheel"}, {"name": "jsonschema", "version": "3.2.0", "latest_version": "4.23.0", "latest_filetype": "wheel"}, {"name": "keyring", "version": "18.0.1", "latest_version": "25.5.0", "latest_filetype": "wheel"}, {"name": "launchpadlib", "version": "1.10.13", "latest_version": "2.1.0", "latest_filetype": "wheel"}, {"name": "lazr.restfulclient", "version": "0.14.2", "latest_version": "0.14.6", "latest_filetype": "wheel"}, {"name": "lazr.uri", "version": "1.0.3", "latest_version": "1.0.7", "latest_filetype": "wheel"}, {"name": "MarkupSafe", "version": "1.1.0", "latest_version": "2.1.5", "latest_filetype": "wheel"}, {"name": "more-itertools", "version": "4.2.0", "latest_version": "10.5.0", "latest_filetype": "wheel"}, {"name": "netifaces", "version": "0.10.4", "latest_version": "0.11.0", "latest_filetype": "wheel"}, {"name": "oauthlib", "version": "3.1.0", "latest_version": "3.2.2", "latest_filetype": "wheel"}, {"name": "pexpect", "version": "4.6.0", "latest_version": "4.9.0", "latest_filetype": "wheel"}, {"name": "pip", "version": "20.0.2", "latest_version": "25.0.1", "latest_filetype": "wheel"}, {"name": "pyasn1", "version": "0.4.2", "latest_version": "0.6.1", "latest_filetype": "wheel"}, {"name": "pyasn1-modules", "version": "0.2.1", "latest_version": "0.4.2", "latest_filetype": "wheel"}, {"name": "PyGObject", "version": "3.36.0", "latest_version": "3.48.2", "latest_filetype": "sdist"}, {"name": "PyHamcrest", "version": "1.9.0", "latest_version": "2.1.0", "latest_filetype": "wheel"}, {"name": "PyJWT", "version": "1.7.1", "latest_version": "2.9.0", "latest_filetype": "wheel"}, {"name": "PyNaCl", "version": "1.3.0", "latest_version": "1.5.0", "latest_filetype": "wheel"}, {"name": "pyOpenSSL", "version": "19.0.0", "latest_version": "25.0.0", "latest_filetype": "wheel"}, {"name": "pyrsistent", "version": "0.15.5", "latest_version": "0.20.0", "latest_filetype": "wheel"}, {"name": "pyserial", "version": "3.4", "latest_version": "3.5", "latest_filetype": "wheel"}, {"name": "python-debian", "version": "0.1.36+ubuntu1.1", "latest_version": "1.0.1", "latest_filetype": "wheel"}, {"name": "PyYAML", "version": "5.3.1", "latest_version": "6.0.2", "latest_filetype": "wheel"}, {"name": "requests", "version": "2.22.0", "latest_version": "2.32.3", "latest_filetype": "wheel"}, {"name": "requests-unixsocket", "version": "0.2.0", "latest_version": "0.3.0", "latest_filetype": "wheel"}, {"name": "SecretStorage", "version": "2.3.1", "latest_version": "3.3.3", "latest_filetype": "wheel"}, {"name": "service-identity", "version": "18.1.0", "latest_version": "24.2.0", "latest_filetype": "wheel"}, {"name": "setuptools", "version": "45.2.0", "latest_version": "75.3.2", "latest_filetype": "wheel"}, {"name": "simplejson", "version": "3.16.0", "latest_version": "3.20.1", "latest_filetype": "wheel"}, {"name": "six", "version": "1.14.0", "latest_version": "1.17.0", "latest_filetype": "wheel"}, {"name": "ssh-import-id", "version": "5.10", "latest_version": "5.11", "latest_filetype": "sdist"}, {"name": "systemd-python", "version": "234", "latest_version": "235", "latest_filetype": "sdist"}, {"name": "Twisted", "version": "18.9.0", "latest_version": "24.11.0", "latest_filetype": "wheel"}, {"name": "urllib3", "version": "1.25.8", "latest_version": "2.2.3", "latest_filetype": "wheel"}, {"name": "wadllib", "version": "1.3.3", "latest_version": "2.0.0", "latest_filetype": "wheel"}, {"name": "wheel", "version": "0.34.2", "latest_version": "0.45.1", "latest_filetype": "wheel"}, {"name": "zipp", "version": "1.0.0", "latest_version": "3.20.2", "latest_filetype": "wheel"}, {"name": "zope.interface", "version": "4.7.1", "latest_version": "7.2", "latest_filetype": "wheel"}]
