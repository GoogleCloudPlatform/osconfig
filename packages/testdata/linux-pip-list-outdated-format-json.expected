[]*packages.PkgInfo{
    &packages.PkgInfo{
        Name:    "attrs",
        Arch:    "all",
        RawArch: "",
        Version: "25.3.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "Automat",
        Arch:    "all",
        RawArch: "",
        Version: "24.8.1",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "blinker",
        Arch:    "all",
        RawArch: "",
        Version: "1.8.2",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "certifi",
        Arch:    "all",
        RawArch: "",
        Version: "2025.4.26",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "chardet",
        Arch:    "all",
        RawArch: "",
        Version: "5.2.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "Click",
        Arch:    "all",
        RawArch: "",
        Version: "8.1.8",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "colorama",
        Arch:    "all",
        RawArch: "",
        Version: "0.4.6",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "configobj",
        Arch:    "all",
        RawArch: "",
        Version: "5.0.9",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "constantly",
        Arch:    "all",
        RawArch: "",
        Version: "23.10.4",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "cryptography",
        Arch:    "all",
        RawArch: "",
        Version: "44.0.3",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "dbus-python",
        Arch:    "all",
        RawArch: "",
        Version: "1.4.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "distro",
        Arch:    "all",
        RawArch: "",
        Version: "1.9.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "distro-info",
        Arch:    "all",
        RawArch: "",
        Version: "1.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "entrypoints",
        Arch:    "all",
        RawArch: "",
        Version: "0.4",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "httplib2",
        Arch:    "all",
        RawArch: "",
        Version: "0.22.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "hyperlink",
        Arch:    "all",
        RawArch: "",
        Version: "21.0.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "idna",
        Arch:    "all",
        RawArch: "",
        Version: "3.10",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "importlib-metadata",
        Arch:    "all",
        RawArch: "",
        Version: "8.5.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "incremental",
        Arch:    "all",
        RawArch: "",
        Version: "24.7.2",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "Jinja2",
        Arch:    "all",
        RawArch: "",
        Version: "3.1.6",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "jsonpatch",
        Arch:    "all",
        RawArch: "",
        Version: "1.33",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "jsonpointer",
        Arch:    "all",
        RawArch: "",
        Version: "3.0.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "jsonschema",
        Arch:    "all",
        RawArch: "",
        Version: "4.23.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "keyring",
        Arch:    "all",
        RawArch: "",
        Version: "25.5.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "launchpadlib",
        Arch:    "all",
        RawArch: "",
        Version: "2.1.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "lazr.restfulclient",
        Arch:    "all",
        RawArch: "",
        Version: "0.14.6",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "lazr.uri",
        Arch:    "all",
        RawArch: "",
        Version: "1.0.7",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "MarkupSafe",
        Arch:    "all",
        RawArch: "",
        Version: "2.1.5",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "more-itertools",
        Arch:    "all",
        RawArch: "",
        Version: "10.5.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "netifaces",
        Arch:    "all",
        RawArch: "",
        Version: "0.11.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "oauthlib",
        Arch:    "all",
        RawArch: "",
        Version: "3.2.2",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "pexpect",
        Arch:    "all",
        RawArch: "",
        Version: "4.9.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "pip",
        Arch:    "all",
        RawArch: "",
        Version: "25.0.1",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "pyasn1",
        Arch:    "all",
        RawArch: "",
        Version: "0.6.1",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "pyasn1-modules",
        Arch:    "all",
        RawArch: "",
        Version: "0.4.2",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "PyGObject",
        Arch:    "all",
        RawArch: "",
        Version: "3.48.2",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "PyHamcrest",
        Arch:    "all",
        RawArch: "",
        Version: "2.1.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "PyJWT",
        Arch:    "all",
        RawArch: "",
        Version: "2.9.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "PyNaCl",
        Arch:    "all",
        RawArch: "",
        Version: "1.5.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "pyOpenSSL",
        Arch:    "all",
        RawArch: "",
        Version: "25.0.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "pyrsistent",
        Arch:    "all",
        RawArch: "",
        Version: "0.20.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "pyserial",
        Arch:    "all",
        RawArch: "",
        Version: "3.5",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "python-debian",
        Arch:    "all",
        RawArch: "",
        Version: "1.0.1",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "PyYAML",
        Arch:    "all",
        RawArch: "",
        Version: "6.0.2",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "requests",
        Arch:    "all",
        RawArch: "",
        Version: "2.32.3",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "requests-unixsocket",
        Arch:    "all",
        RawArch: "",
        Version: "0.3.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "SecretStorage",
        Arch:    "all",
        RawArch: "",
        Version: "3.3.3",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "service-identity",
        Arch:    "all",
        RawArch: "",
        Version: "24.2.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "setuptools",
        Arch:    "all",
        RawArch: "",
        Version: "75.3.2",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "simplejson",
        Arch:    "all",
        RawArch: "",
        Version: "3.20.1",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "six",
        Arch:    "all",
        RawArch: "",
        Version: "1.17.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "ssh-import-id",
        Arch:    "all",
        RawArch: "",
        Version: "5.11",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "systemd-python",
        Arch:    "all",
        RawArch: "",
        Version: "235",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "Twisted",
        Arch:    "all",
        RawArch: "",
        Version: "24.11.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "urllib3",
        Arch:    "all",
        RawArch: "",
        Version: "2.2.3",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "wadllib",
        Arch:    "all",
        RawArch: "",
        Version: "2.0.0",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "wheel",
        Arch:    "all",
        RawArch: "",
        Version: "0.45.1",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "zipp",
        Arch:    "all",
        RawArch: "",
        Version: "3.20.2",
        Source:  packages.Source{},
    },
    &packages.PkgInfo{
        Name:    "zope.interface",
        Arch:    "all",
        RawArch: "",
        Version: "7.2",
        Source:  packages.Source{},
    },
}